query-builder/src/main/java/com/datastax/oss/driver/api/querybuilder/relation/Relation.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.querybuilder.relation;
+
+import com.datastax.oss.driver.api.core.CqlIdentifier;
+import com.datastax.oss.driver.internal.querybuilder.relation.DefaultColumnComponentRelationBuilder;
+import com.datastax.oss.driver.internal.querybuilder.relation.DefaultColumnRelationBuilder;
+import com.datastax.oss.driver.internal.querybuilder.relation.DefaultTokenRelationBuilder;
+import com.datastax.oss.driver.internal.querybuilder.relation.RawRelation;
+import com.google.common.collect.Iterables;
+import java.util.Arrays;
+
+/** A relation in a WHERE clause. */
+public interface Relation {
+
+  /**
+   * Builds an arbitrary relation from a raw string.
+   *
+   * <p>The contents be appended to the query as-is, without any syntax checking or escaping. This
+   * method should be used with caution, as it's possible to generate invalid CQL that will fail at
+   * execution time; on the other hand, it can be used as an "escape hatch" to handle edge cases
+   * that are not covered by the query builder.
+   */
+  static Relation isRaw(String raw) {
+    return new RawRelation(raw);
+  }
+
+  /**
+   * Builds a relation testing a column.
+   *
+   * <p>This call must be chained with an operator, for example:
+   *
+   * <pre>{@code
+   * selectFrom("foo").all().where(isColumn("k").eq(bindMarker()));
+   * }</pre>
+   */
+  static ColumnRelationBuilder isColumn(CqlIdentifier id) {
```

Comment:
We need to name this differently than selectors to avoid static import clashes:

```java
selectFrom("foo")
  .selectors(column("k"))
  .where(isColumn("k").eq(bindMarker()))
  .build(false)
```